core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1159,6 +1161,9 @@ class LogManager(logDirs: Seq[File],
           checkpointLogStartOffsetsInDir(logDir, logsToCheckpoint)
         }
         addLogToBeDeleted(removedLog)
+        if (deleteRemote && removedLog.remoteLogEnabled())
```

Comment:
It's still weird to reference remoteLogEnabled in LogManager since it only manages local data.

Reply:
I moved this code into `ReplicaManager`